homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -73,6 +73,13 @@ def convert_and_round(
         icon="mdi:ev-station",
         value=lambda x, y: x.value,
     ),
+    "charging_target": BMWSensorEntityDescription(
+        key="charging_target",
+        name="Charging target",
+        key_class="fuel_and_battery",
+        icon="mdi:battery-charging-90",
+        device_class=SensorDeviceClass.BATTERY,
```

Comment:
I would remove the device_class as it's a setting and not representing the actual state of the battery.
Instead you can add `unit_type=PERCENTAGE,` and change the icon to `icon="mdi:battery-charging-high",`

Reply:
Agree, this is not the state of the battery. It's a target.

Reply:
Thanks. Was unsure what this should be - changed as requested